Userland/Libraries/LibWeb/FileAPI/Blob.cpp

```diff
@@ -221,66 +221,85 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<Blob>> Blob::construct_impl(JS::Realm& real
 
 // https://w3c.github.io/FileAPI/#dfn-slice
 WebIDL::ExceptionOr<JS::NonnullGCPtr<Blob>> Blob::slice(Optional<i64> start, Optional<i64> end, Optional<String> const& content_type)
+{
+    // 1. Let sliceStart, sliceEnd, and sliceContentType be null.
+    // 2. If start is given, set sliceStart to start.
+    // 3. If end is given, set sliceEnd to end.
+    // 3. If contentType is given, set sliceContentType to contentType.
+    // 4. Return the result of slice blob given this, sliceStart, sliceEnd, and sliceContentType.
+    return slice_blob(start, end, content_type);
+}
+
+// https://w3c.github.io/FileAPI/#slice-blob
+WebIDL::ExceptionOr<JS::NonnullGCPtr<Blob>> Blob::slice_blob(Optional<i64> start, Optional<i64> end, Optional<String> const& content_type)
 {
     auto& vm = realm().vm();
 
-    // 1. The optional start parameter is a value for the start point of a slice() call, and must be treated as a byte-order position, with the zeroth position representing the first byte.
-    //    User agents must process slice() with start normalized according to the following:
+    // 1. Let originalSize be blob’s size.
+    auto original_size = size();
+
+    // 2. The start parameter, if non-null, is a value for the start point of a slice blob call, and must be treated as a byte-order position,
+    //    with the zeroth position representing the first byte. User agents must normalize start according to the following:
     i64 relative_start;
     if (!start.has_value()) {
-        // a. If the optional start parameter is not used as a parameter when making this call, let relativeStart be 0.
+        // a. If start is null, let relativeStart be 0.
         relative_start = 0;
     } else {
         auto start_value = start.value();
-        // b. If start is negative, let relativeStart be max((size + start), 0).
+
+        // b. If start is negative, let relativeStart be max((originalSize + start), 0).
         if (start_value < 0) {
-            relative_start = max((size() + start_value), 0);
+            relative_start = max((original_size + start_value), 0);
```

Comment:
The type of the first argument is getting converted to `u64` here (the type of `original_size`). This can be fixed with a cast here, or you could change the type of `original_size` to `i64`.